### PR TITLE
Go 1.11 modules support

### LINKED ISSUE #44: Go 1.11 modules: go-resty/resty import path
As per go-resty/resty#178, the import path should be `import "gopkg.in/resty.v1"` instead of `github.com/go-resty/resty` which is being used right now. It causes problem with Go 1.11 modules system with following error:

```
go: github.com/go-resty/resty@v1.9.1: parsing go.mod: unexpected module path "gopkg.in/resty.v1"
go: error loading module requirements
```

It would be great if this is fixed.

### PR DESCRIPTION
**Pull request by maddie**
Fixes #44

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ go:
   - tip
 
 install:
+  - go get -u golang.org/x/vgo
```
Comment (tamalsaha) [inline .travis.yml]: Why do you need this with Go 1.11?

Comment (displague) [reply .travis.yml]: Thanks for that -- looking at https://dave.cheney.net/2018/07/16/using-go-modules-with-travis-ci now

Comment (tamalsaha) [reply .travis.yml]: https://roberto.selbach.ca/intro-to-go-modules/ is the best user-guide I have found.

I converted one of our projects to use Go 1.11 with travis. See here: https://github.com/appscode/go/commit/97214a81824c6834bc34a0a0d5f1e17ed7099ef9

**Comment by displague** [comment]
@maddie I've been pushing commits on this branch trying to get a pure `go mod` build.  Still failing on the travis end.

**Comment by maddie** [comment]
I'm not quite familiar with Travis, I'll see what I can do

**Comment by maddie** [comment]
It's possible that alecthomas/gometalinter#521 causes the failure

**Comment by maddie** [comment]
Running `make test ARGS='-coverprofile=coverage.txt -covermode=atomic'` or `make lint-warn` with the new `Makefile` fails due to the above mentioned problem:

Output:
```
$ make test ARGS='-coverprofile=coverage.txt -covermode=atomic'
go build ./...
go get -u gopkg.in/alecthomas/gometalinter.v2
# gopkg.in/alecthomas/gometalinter.v2
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:36:94: cannot use loadConfig (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("config", "Load JSON configuration from file.").Envar("GOMETALINTER_CONFIG").Action
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:38:100: cannot use disableAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("disable", "Disable previously enabled linters.").PlaceHolder("LINTER").Short('D').Action
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:39:99: cannot use enableAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("enable", "Enable previously disabled linters.").PlaceHolder("LINTER").Short('E').Action
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:40:83: cannot use cliLinterOverrides (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("linter", "Define a linter.").PlaceHolder("NAME:COMMAND:PATTERN").Action
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:43:56: cannot use disableAllAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("disable-all", "Disable all linters.").Action
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:44:54: cannot use enableAllAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("enable-all", "Enable all linters.").Action
../../../go/pkg/mod/gopkg.in/alecthomas/gometalinter.v2@v2.0.8/main.go:200:12: cannot use loadDefaultConfig (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.Action
make: *** [/bin/gometalinter.v2] Error 2
```

So the solutions are:

- workarounds in `gometalinter`'s issue tracker, which are not perfect
- disable linter temporarily
- wait for `gometalinter` to fix compatibility with Go 1.11 modules

**Comment by maddie** [comment]
Now as the author of `gometalinter` said Go 1.11 modules support will not be coming anytime soon, I guess option 3 is out of consideration.

**Comment by displague** [comment]
https://github.com/alecthomas/gometalinter/issues/531 might be an approach

**Comment by cliedeman** [comment]
I wonder if go mod vendor could get the tests to pass.

**Pull request merged**